### DISCUSSION TITLE
Unable to access environment variables in local environment

### DISCUSSION THREAD

**Discussion by creativebyte-io** [vercel/next.js#22101] [Help]
Hi,

I am trying to store CLIENT_SECRET in environment variables and use it to make api calls in getServerSideProps. I was trying to find the most ideal way to do this.

I am using NextJS 10.0.1. The documentation says that we could store the variables in .env.local and access them in the code in the local environment.

However, the variable is undefined. I even tried doing it through next.config.js like this:

```
//next.config.js
module.exports = {
    serverRuntimeConfig: {
        client_secret : process.env.client_secret,
      }
 }
```

```
//.env.local
client_secret="myclientsecret"
```

```
//page that has getServerSideProps
import getConfig from 'next/config'
const { serverRuntimeConfig } = getConfig()
getServerSideProps(params){
    console.log(serverRuntimeConfig.client_secret) //undefined
    console.log(process.env.client_secret) //undefined
}
```

What could I be doing wrong here?
Thanks in advance.

**Comment by krystianjj** [comment]
Try put variable in uppercase `CLIENT_SECRET` and without semicolon.
```
// .env.local
CLIENT_SECRET=myclientsecret
```
and read it
`console.log(process.env.CLIENT_SECRET)`
